distage/distage-framework-docker/src/main/scala/izumi/distage/docker/healthcheck/ContainerHealthCheck.scala

```diff
@@ -3,38 +3,52 @@ package izumi.distage.docker.healthcheck
 import izumi.distage.docker.Docker._
 import izumi.distage.docker.DockerContainer
 import izumi.distage.docker.healthcheck.ContainerHealthCheck.HealthCheckResult
+import izumi.distage.docker.healthcheck.chain.{ChainedContainerHealthCheck, HttpProtocolCheck, PostgreSqlProtocolCheck}
 import izumi.fundamentals.collections.nonempty.NonEmptyList
 import izumi.logstage.api.IzLogger
 
 trait ContainerHealthCheck[Tag] {
   def check(logger: IzLogger, container: DockerContainer[Tag]): HealthCheckResult
-}
 
+  final def combine(next: ChainedContainerHealthCheck[Tag]): ContainerHealthCheck[Tag] = this ++ next
+  final def ++(next: ChainedContainerHealthCheck[Tag]): ContainerHealthCheck[Tag] = {
```

Comment:
Hmm, `ChainedContainerHealthCheck[Tag]` is the same as `AvailablePorts => ContainerHealthCheck[Tag]`, a new trait may be redundant